### PR TITLE
TEC-6509 switch to RestFB DefaultJsonMapper >v2

### PR DESCRIPTION
**Pull request by diepjy**
### Description of Changes

RestFB has been on v2 for quit a while, but the implementation for DefaultJsonMapper in this project is using their V1 implementation. The main update has been the use of minimal-json rather than org.json handling of JSON object. This Pr copies over their new implementation and updates all files that are copied over from RestFB to V2.

### Documentation

https://restfb.com/documentation/

### Risks & Impacts

`DefaultJsonMapper` determines how JSON strings are mapped into Java objects and vice versa.
There's a nice diff of the updates

### Testing

Unit tests pass again!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,7 +55,11 @@
 			<version>0.1.2</version>
 			<scope>runtime</scope>
 		</dependency>
-
+		<dependency>
+			<groupId>com.eclipsesource.minimal-json</groupId>
```
[the remark this answers is not in the document]
Comment (eddspencer) [reply pom.xml]: It is used in the `com.eclipsesource.json` imports used in the mappers, so we are using the minimal-json package here to handle the `String <-> JSONObject` conversion and the `DefaultJsonMapper` handles `String <-> Generic class/enum` conversions.

**Comment by diepjy** [comment]
@eddspencer  not too sure, perhaps when minimal-json constructs the json object from the object

**Comment by eddspencer** [comment]
OK well it wont make a difference to the running of things and that test seems stable now :+1:

**Comment by MarcFletcher** [comment]
I've tried to resolve the conflicts but it wasn't obvious exactly what the new dependencies in the POM are. Please resolve.

**Comment by diepjy** [comment]
@MarcFletcher I've resolve the merge conflicts

**Pull request merged**